### PR TITLE
[Workspace] Redirect "Sample data" link to the page in the same workspace

### PR DESCRIPTION
**Pull request by Kapian1234**
### Description
Redirect "Sample data" link to the page in the same workspace

## Screenshot
workspace enabled

https://github.com/user-attachments/assets/49d705ba-0679-4bf6-a44b-98e94140cef2

workspace disabled

https://github.com/user-attachments/assets/3a28d78d-1e12-407c-a63a-530c0cc3447f

## Testing the changes
`yarn test:jest src/plugins/dashboard/public/application/utils/get_no_items_message.test.tsx`

## Changelog

- fix: Redirect sample data link to the page in the same workspace

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/dashboard/public/application/utils/get_no_items_message.tsx [hunk 1]
```diff
@@ -60,11 +62,9 @@ export const getNoItemsMessage = (
                 values={{
                   sampleDataInstallLink: (
                     <EuiLink
-                      onClick={() =>
-                        application.navigateToApp('home', {
```
Comment (wanglam) [inline src/plugins/dashboard/public/application/utils/get_no_items_message.tsx:64]: What about call `application.navigationToApp('import_sample_data')` directly? Then we won't need so much modifications.

Comment (Kapian1234) [reply src/plugins/dashboard/public/application/utils/get_no_items_message.tsx:64]: Oh I didn't know that. It's a much better way to do that, thanks!

File: src/plugins/dashboard/public/application/utils/get_no_items_message.test.tsx [hunk 2]
```diff
@@ -6,9 +6,9 @@
 import { getNoItemsMessage } from './get_no_items_message';
 import { mountWithIntl } from 'test_utils/enzyme_helpers';
 import { ApplicationStart, Capabilities, PublicAppInfo } from 'opensearch-dashboards/public';
-import { EuiLink } from '@elastic/eui';
-import { RecursiveReadonly } from '@osd/utility-types';
 import { Observable } from 'rxjs';
+import { RecursiveReadonly } from '@osd/utility-types';
+import { EuiLink } from '@elastic/eui';
```
Comment (wanglam) [inline src/plugins/dashboard/public/application/utils/get_no_items_message.test.tsx:11]: Nit: Seems these import changes are not necessary, prefer to keep the original code.

**Comment by SuZhou-Joe** [comment]
@Kapian1234 Can we post a record on 2 scenarios:
1. Workspace enabled and click the sample data link inside workspace
2. Workspace disabled and click the sample data link outside workspace.

**Comment by Kapian1234** [comment]
> @Kapian1234 Can we post a record on 2 scenarios:
> 
> 1. Workspace enabled and click the sample data link inside workspace
> 2. Workspace disabled and click the sample data link outside workspace.

ok

> @Kapian1234 Can we post a record on 2 scenarios:
> 
> 1. Workspace enabled and click the sample data link inside workspace
> 2. Workspace disabled and click the sample data link outside workspace.

Added

**Review by ruanyl — DISMISSED** [review]

**Comment by ruanyl** [comment]
@Kapian1234 snapshot test failed, please take a look.

**Comment by Kapian1234** [comment]
> @Kapian1234 snapshot test failed, please take a look.

ok

**Comment by Kapian1234** [comment]
> @Kapian1234 snapshot test failed, please take a look.

But it runs successfully on my local

**Comment by ruanyl** [comment]
> > @Kapian1234 snapshot test failed, please take a look.
> 
> But it runs successfully on my local

Running this failed on my local
```
yarn test:jest src/plugins/dashboard/public/application/utils/get_no_items_message.test.tsx
```

**Review by ruanyl — APPROVED** [review]

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**